localstripe/localstripe-v3.js

```diff
@@ -226,13 +226,13 @@ Stripe = (apiKey) => {
             payment_method_data: {
               type: 'card',
               card: {
-                number: element.value.number,
-                exp_month: element.value.exp_month,
-                exp_year: element.value.exp_year,
-                cvc: element.value.cvc,
+                number: data.payment_method.card.value.number,
+                exp_month: data.payment_method.card.value.exp_month,
+                exp_year: data.payment_method.card.value.exp_year,
+                cvc: data.payment_method.card.value.cvc,
               },
               billing_details: {
-                address: {postal_code: element.value.address_zip},
+                address: {postal_code: data.payment_method.card.value.address_zip},
```

Comment:
From my understanding of the new `confirmCardSetup()` API, the `data` argument doesn't have a `.payment_method.card.value.address_zip`. Instead, it should be called with `.payment_method` being an "object containing data to create a PaymentMethod with", like in https://stripe.com/docs/api/payment_methods/create.

It would mean:
- This code would crash if called from code that supports the new Stripe arguments?
- This seems to go in favor of my comment just above!
- Modifications are probably needed in the *card mounting* code, above in the file.
- Modifications are probably need in legacy `handleCardSetup()`, so that old code keeps worknig.

Reply:
Hey @adrienverge, sorry but it have taken me forever to only start to understand.
What you mean is that you think they have change the object returned by https://stripe.com/docs/js/elements_object/create_element?type=card (and properties added in it by `mount`) in the same time they depreciate `handleCardSetup()`?

Reply:
(I know, it also took me a long time (reading the docs) to detect this potential problem!)

No, I'm saying that when `Stripe().confirmCardSetup()` is called from a valid JS app, it is now called with a `data` argument (no more `Element`/card object passed as direct argument), and this `data` argument doesn't seem to contain `data.payment_method.card.value.address_zip` (correct me if I'm wrong, but this is what I understand from the docs).

So, the function as it is written now, won't work with:

```js
// example from the docs
Stripe().confirmCardSetup(clientSecret, {
  payment_method: {
    card: cardElement,
    billing_details: {
      address: {
        postal_code: '12345',
      },
    },
  },
});
```

neither with:

```js
// example from the docs
Stripe().confirmCardSetup(clientSecret, {
  payment_method: {
    type: "card",
    card: {
      "number": "4242424242424242",
      "exp_month": 12,
      "exp_year": 2020,
      "cvc": "314",
    },
    billing_details: {
      address: {
        postal_code: '12345',
      },
    },
  },
});
```

because it will:
- ignore `billing_details.address.postal_code`
- crash on `data.payment_method.card.value.address_zip` (actually it won't crash today, because of the old *card mounting code*, cf. my previous comment).

Do you confirm my reading of the documentation?

Reply:
Hey Adrien.
I have look more into the `Element` object, in my console using the real Stripe API and inside [@type/stripe-v3](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/stripe-v3/index.d.ts#L671-L685).
For me all the variables we use here to simulate the API (`card.value.number`, `card.value.exp_year`, etc) don't exists at all in real situation.
Thus, it can't:
- lead to a crash
- make old code unworking.
No?
I can simplify the object so it better glue with the final http fetch to the http API, but in another commit/PR, no?